### PR TITLE
Re-enable HeapAttackIT

### LINKED ISSUE #100678: [CI] HeapAttackIT testManyConcat failing
**Build scan:**
https://gradle-enterprise.elastic.co/s/z27lugpake4um/tests/:x-pack:plugin:esql:qa:server:single-node:javaRestTest/org.elasticsearch.xpack.esql.qa.single_node.HeapAttackIT/testManyConcat
**Reproduction line:**
```
./gradlew ':x-pack:plugin:esql:qa:server:single-node:javaRestTest' --tests "org.elasticsearch.xpack.esql.qa.single_node.HeapAttackIT.testManyConcat" -Dtests.seed=6986B005E1B4E2DE -Dtests.locale=vi-VN -Dtests.timezone=America/Lima -Druntime.java=17
```

**Applicable branches:**
main

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.esql.qa.single_node.HeapAttackIT&tests.test=testManyConcat
**Failure excerpt:**
```
java.lang.OutOfMemoryError: Java heap space

  at __randomizedtesting.SeedInfo.seed([6986B005E1B4E2DE:10470D421C8F6C49]:0)
  at org.apache.http.util.CharArrayBuffer.expand(CharArrayBuffer.java:60)
  at org.apache.http.util.CharArrayBuffer.append(CharArrayBuffer.java:90)
  at org.apache.http.util.EntityUtils.toString(EntityUtils.java:228)
  at org.apache.http.util.EntityUtils.toString(EntityUtils.java:308)
  at org.elasticsearch.xpack.esql.qa.single_node.HeapAttackIT.testManyConcat(HeapAttackIT.java:184)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)

```

### LINKED ISSUE #103527: [CI] HeapAttackIT classMethod failing
**Build scan:**
https://gradle-enterprise.elastic.co/s/pbsnlozkeubg6/tests/:x-pack:plugin:esql:qa:server:heap-attack:javaRestTest/org.elasticsearch.xpack.esql.qa.heap_attack.HeapAttackIT

**Reproduction line:**
```
null
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
[Failure dashboard for `org.elasticsearch.xpack.esql.qa.heap_attack.HeapAttackIT#classMethod`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('classMethod'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.qa.heap_attack.HeapAttackIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure excerpt:**
```
java.lang.Exception: Suite timeout exceeded (>= 1200000 msec).

  at __randomizedtesting.SeedInfo.seed([BE1188EC2E762181]:0)

```

### PR DESCRIPTION
**Pull request by dnhatn**
This PR enables ESQL heap attack tests. I have run this suite over 500 iterations with different hardware configurations for the last two days, and all have been successful with the changes in https://github.com/elastic/elasticsearch/pull/104159. 

Additionally, this PR adds an action that can trigger OOM to generate a heap dump if a test takes more than 5 minutes. I've seen cases (previously with our CI) where the test didn't result in OOM but was taking too long. Having the ability to inspect the heap in such cases would be beneficial.

Closes #103527
Closes #100678

### CODE REVIEW

File: test/external-modules/delayed-aggs/build.gradle [hunk 1]
```diff
@@ -5,6 +5,13 @@
  * in compliance with, at your election, the Elastic License 2.0 or the Server
  * Side Public License, v 1.
  */
+import org.elasticsearch.gradle.internal.info.BuildParams
+
+apply plugin: 'elasticsearch.legacy-yaml-rest-test'
```
Comment (dnhatn) [inline test/external-modules/delayed-aggs/build.gradle:10]: This was moved from the parent module.

File: test/external-modules/error-query/build.gradle [hunk 2]
```diff
@@ -6,6 +6,13 @@
  * Side Public License, v 1.
  */
 
+import org.elasticsearch.gradle.internal.info.BuildParams
+apply plugin: 'elasticsearch.legacy-yaml-rest-test'
```
Comment (dnhatn) [inline test/external-modules/error-query/build.gradle:10]: This also was moved from the parent module.

**Comment by dnhatn** [comment]
test this please

**Comment by dnhatn** [comment]
run elasticsearch-ci/check-serverless-submodule

**Review by mark-vieira — APPROVED** [review]

File: test/external-modules/esql-heap-attack/src/main/java/org/elasticsearch/test/esql/heap_attack/RestTriggerOutOfMemoryAction.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        return List.of(new Route(POST, "/_trigger_out_of_memory"));
+    }
+
+    @Override
+    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) {
+        LOGGER.error("triggering out of memory");
+        List<int[]> values = new ArrayList<>();
+        return channel -> {
+            while (true) {
+                values.add(new int[1024 * 1024]);
```
Comment (nik9000) [inline test/external-modules/esql-heap-attack/src/main/java/org/elasticsearch/test/esql/heap_attack/RestTriggerOutOfMemoryAction.java:48]: Rude.

Comment (nik9000) [reply test/external-modules/esql-heap-attack/src/main/java/org/elasticsearch/test/esql/heap_attack/RestTriggerOutOfMemoryAction.java:48]: Wonderful.

**Review by nik9000 — APPROVED** [review]

**Comment by dnhatn** [comment]
@mark-vieira @nik9000 Thanks for reviews.

**Pull request merged**